code/ui/components/src/components/Button/Button.tsx

```diff
@@ -1,272 +1,216 @@
-    const color = theme.color.primary;
+    if (
+      props.small ||
+      props.isLink ||
+      props.primary ||
+      props.secondary ||
+      props.tertiary ||
+      props.gray ||
+      props.outline ||
+      props.inForm ||
+      props.containsIcon
+    ) {
+      const buttonContent = React.Children.toArray(props.children).filter(
+        (e) => typeof e === 'string' && e !== ''
+      );
 
-    return outline && primary
-      ? {
-          boxShadow: `${color} 0 0 0 1px inset`,
-          color,
+      deprecate(
+        `Use of deprecated props in the button ${
+          buttonContent.length > 0 ? `"${buttonContent[0]}"` : 'component'
+        } detected, see the migration notes at https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#new-ui-and-props-for-button-and-iconbutton-components`
+      );
+    }
 
-          'svg path:not([fill])': {
-            fill: color,
-          },
+    return (
+      <StyledButton
+        as={Comp}
+        ref={ref}
+        variant={lovalVariant}
+        size={localSize}
+        padding={padding}
+        disabled={disabled}
+        active={active}
+        animating={isAnimating}
+        animation={animation}
+        onClick={handleClick}
+        {...props}
+      />
+    );
+  }
+);
 
-          '&:hover': {
-            boxShadow: `${color} 0 0 0 1px inset`,
-            background: 'transparent',
-          },
+Button.displayName = 'Button';
 
-          '&:active': {
-            background: color,
-            boxShadow: `${color} 0 0 0 1px inset`,
-            color: theme.color.tertiary,
-          },
-          '&:focus': {
-            boxShadow: `${color} 0 0 0 1px inset, ${rgba(color, 0.4)} 0 1px 9px 2px`,
-            outline: 'none',
-          },
-          '&:focus:hover': {
-            boxShadow: `${color} 0 0 0 1px inset, ${rgba(color, 0.2)} 0 8px 18px 0px`,
-          },
-        }
-      : {};
+const StyledButton = styled('button', {
+  shouldForwardProp: (prop) => isPropValid(prop),
+})<
+  ButtonProps & {
+    animating: boolean;
+    animation: ButtonProps['animation'];
+  }
+>(({ theme, variant, size, disabled, active, animating, animation, padding }) => ({
+  border: 0,
+  cursor: disabled ? 'not-allowed' : 'pointer',
+  display: 'inline-flex',
+  gap: '6px',
+  alignItems: 'center',
+  justifyContent: 'center',
+  overflow: 'hidden',
+  padding: (() => {
+    if (padding === 'small' && size === 'small') return '0 7px';
+    if (padding === 'small' && size === 'medium') return '0 9px';
+    if (size === 'small') return '0 10px';
+    if (size === 'medium') return '0 12px';
+    return 0;
+  })(),
+  height: size === 'small' ? '28px' : '32px',
+  position: 'relative',
+  textAlign: 'center',
+  textDecoration: 'none',
+  transitionProperty: 'background, box-shadow',
+  transitionDuration: '150ms',
+  transitionTimingFunction: 'ease-out',
+  verticalAlign: 'top',
+  whiteSpace: 'nowrap',
+  userSelect: 'none',
+  opacity: disabled ? 0.5 : 1,
```

Comment:
this way of setting something to be disabled is a bit unfortunate. if the button is on top of something that has multiple colors, it will actually be semi-transparent, where I think what we really want is the colors to be slighlty muted?

Reply:
I agree with you on this one but I think this is something we might change after the theming v2.0 project. For now there is pretty low risk that the button will be below something else.